### ISSUE TITLE
[Feature Request] Annotations: add dynamic step to show many annotations as a range

### ISSUE THREAD

**Issue by bohandley** [grafana/grafana#55187]
A response to https://github.com/grafana/support-escalations/issues/3120
**Scope:** Grafana annotations
**Impact:** Users who create annotations for ranges and zoom in and out of panels
**Why now?** This is a visual feature. The feature will help users who do not wish to change the min step of annotations every time the zoom on a panel.

Description: Add a dynamic min step option in creating an annotation.

We want to show annotations in a range and not have them broken down into individual elements when changing the panel size by zooming out. When an annotation returns multiple events, Prometheus can either return a range of events or each individual event. 

Expected Result:
![Image](https://user-images.githubusercontent.com/25674746/190228236-2979b597-991b-452c-896f-09577bf2f14c.png)

Undesired:
![Image](https://user-images.githubusercontent.com/25674746/190228255-f932d740-cdab-4c02-8b37-bfcff127f0a5.png)

This is dependent on the min step chosen by the person who created the annotation. 

![Image](https://user-images.githubusercontent.com/25674746/190228369-78c5c11e-f595-4959-b9c6-1a52680e3ed8.png)

We compare the last event + the step to the next event's timestamp to either show the range or the many elements
https://github.com/grafana/grafana/blob/1e06b0170b3145ffd53b5fcb0d16d710c6bfc42f/public/app/plugins/datasource/prometheus/datasource.tsx#L808-L812

Acceptance Criteria: Show the range of annotations when a user chooses the dynamic min step.

Things to consider: 
It is not suggested to use this feature when creating annotations for the purpose of alerting
https://github.com/grafana/support-escalations/issues/2019#

Fixes https://github.com/grafana/grafana/issues/63687

**Comment by rs** [comment]
I don't think it is practical. Whatever value you use, you will either get the issue with larger window or will lose granularity with smaller windows.

**Comment by leeoniya** [comment]
@bohandley @gtk-grafana this feels like a dataviz feature, basically annotations clustering that woupd be datasource agnostic?

we're mostly finished rewriting the annotation rendering and editing plugin in all panels now. do you want us to move it to our backlog, or is there something prom-specific here?

**Comment by bohandley** [comment]
This is not Prometheus specific but was a special thing introduced into Prometheus to do what your squad is working on. @leeoniya do you have a PR link to that work?

**Comment by bohandley** [comment]
Moving this to dataviz as offered by @leeoniya, thank you!

**Comment by bohandley** [comment]
Closing this because this feature request was originally created to solve [this issue](https://github.com/grafana/grafana/issues/63687), which was recently solved by [this PR](https://github.com/grafana/grafana/pull/97734).
